src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift

```diff
@@ -141,93 +248,121 @@ class DiaryOverviewDayTableViewCell: UITableViewCell {
-		imageView.contentMode = .center
-		imageView.image = test.result == .negative ? UIImage(imageLiteralResourceName: "Test_green") : UIImage(imageLiteralResourceName: "Test_red")
-
-		let titleLabel = ENALabel()
-		titleLabel.style = .body
-		titleLabel.text = test.type == .pcr ? AppStrings.ContactDiary.Overview.Tests.pcrRegistered : AppStrings.ContactDiary.Overview.Tests.antigenDone
-
-		let detailLabel = ENALabel()
-		detailLabel.style = .subheadline
-		detailLabel.textColor = .enaColor(for: .textPrimary2)
-		detailLabel.text = test.result == .negative ? AppStrings.ContactDiary.Overview.Tests.negativeResult : AppStrings.ContactDiary.Overview.Tests.positiveResult
-
-		let verticalStackView = UIStackView(arrangedSubviews: [titleLabel, detailLabel])
-		verticalStackView.axis = .vertical
-		verticalStackView.spacing = 8.0
-
-		let horizontalStackView = UIStackView(arrangedSubviews: [imageView, verticalStackView])
-		horizontalStackView.translatesAutoresizingMaskIntoConstraints = false
-		horizontalStackView.alignment = .center
-		horizontalStackView.spacing = 15.0
-		containerView.addSubview(horizontalStackView)
-
-		NSLayoutConstraint.activate(
-			[
-				separator.heightAnchor.constraint(equalToConstant: 1.0),
-				separator.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
-				separator.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+		bottomBackground.layer.cornerRadius = 14
+		if #available(iOS 13.0, *) {
+			bottomBackground.layer.cornerCurve = .continuous
+		}
+		bottomBackground.layer.borderWidth = 1
+		bottomBackground.layer.borderColor = UIColor.enaColor(for: .hairline).cgColor
+		bottomBackground.layer.maskedCorners = [.layerMaxXMaxYCorner, .layerMinXMaxYCorner]
+
+		// Show same background like the topBackground when we have not the encountersVisitsContainerStackView at the bottom displayed or the hole day is empty
+		if (testsStackView.isHidden && exposureHistoryStackView.isHidden && checkinHistoryContainerStackView.isHidden) || !encountersVisitsContainerStackView.isHidden {
+			bottomBackground.backgroundColor = .enaColor(for: .cellBackground)
+		} else {
+			bottomBackground.backgroundColor = .enaColor(for: .darkBackground)
+		}
+	}
 
-				imageView.widthAnchor.constraint(equalToConstant: 32),
-				horizontalStackView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: 18.0),
-				horizontalStackView.topAnchor.constraint(equalTo: containerView.topAnchor, constant: 12.0),
-				horizontalStackView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -8.0),
-				horizontalStackView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor, constant: -8.0)
-			]
-		)
-		return containerView
+	private func drawBorders(to orientations: [BorderOrientation], on view: UIView) {
+		// Prevent drawing the borders mutliple times by setting a tag on it and search for the tag.
+		if view.subviews.contains(where: { $0.tag == view.hashValue }) {
+			return
+		}
```

Comment:
not a nice solution - maybe a better solution could be an inherited view with a property?

Reply:
Yeah, your idea is a better approach - I created a small custom class with a property and check for this property now.